src/main/java/org/jabref/gui/texparser/ParseTexResultViewModel.java

```diff
@@ -0,0 +1,31 @@
+package org.jabref.gui.texparser;
+
+import javafx.beans.property.ReadOnlyListWrapper;
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+
+import org.jabref.gui.AbstractViewModel;
+import org.jabref.model.texparser.Citation;
+import org.jabref.model.texparser.TexParserResult;
+
+public class ParseTexResultViewModel extends AbstractViewModel {
+
+    private final ObservableList<ReferenceWrapper> referenceList;
+    private final ObservableList<Citation> citationList;
+
+    public ParseTexResultViewModel(TexParserResult texParserResult) {
+        this.referenceList = FXCollections.observableArrayList();
+        this.citationList = FXCollections.observableArrayList();
+
+        texParserResult.getCitationsKeySet().forEach(key -> referenceList.add(
```

Comment:
Performancewise it should be better if you directly work with the citations map (instead of keys and then get the citation for it). Maybe even something like `referenceList  = getCitations().stream().map(new ReferenceWrapper(...)))` works.